cloudinit/ssh_util.py

```diff
@@ -321,17 +321,27 @@ def check_create_path(username, filename, strictmodes):
         home_folder = os.path.dirname(user_pwent.pw_dir)
         for directory in directories:
             parent_folder += "/" + directory
-            if home_folder.startswith(parent_folder):
+            if (home_folder.startswith(parent_folder) or
+                    parent_folder == user_pwent.pw_dir):
                 continue
 
             if not os.path.isdir(parent_folder):
```

Comment:
There is one weird case that is not covered by this `if`: `parent_folder` existing but not-a-directory (file, symlink). In this case cloud-init is going to fail creating deeper subdirectories or the authorized_keys file, possibly doing unexpected things in the case of a symlink. Do you think we should cover this case, returning early when `os.path.isdir(parent_folder)` but `os.path.exists(parent_folder)`?

Reply:
More specifically I'm worried by a user doing e.g. `ln -s .ssh /root/.ssh`, and of what would cloud-init do.

Reply:
Pushed https://github.com/canonical/cloud-init/pull/984/commits/240a0795d9484a18b4a5798fb0f08bd5b7c139bf , which I think addresses this.